packages/stryker/src/input/InputFileResolver.ts

```diff
@@ -50,34 +53,77 @@ export default class InputFileResolver {
   }
 
   private resolveInputFiles() {
-    if (this.fileResolver) {
-      return this.fileResolver.resolve();
+    if (this.filePatterns) {
+      return this.expand(this.filePatterns);
     } else {
       return this.resolveFilesUsingGit();
     }
   }
 
-  private resolveFilesUsingGit(): Promise<string[]> {
-    return childProcessAsPromised
-      .exec(
+  private resolveMutateFiles() {
+    return this.expand(this.mutatePatterns, !shallowEquals(this.mutatePatterns, new Config().mutate));
+
+    function shallowEquals<T>(arr1: ReadonlyArray<T>, arr2: ReadonlyArray<T>): boolean {
+      if (arr1.length !== arr2.length) {
+        return false;
+      } else {
+        for (let i = 0; i < arr1.length; i++) {
+          if (arr1[i] !== arr2[i]) {
+            return false;
+          }
+        }
+        return true;
+      }
+    }
+  }
+
+  /**
+   * Takes a list of globbing patterns and expands them into files.
+   * If a patterns starts with a `!`, it negates the pattern.
+   * @param patterns The patterns to expand into files
+   */
+  private async expand(patterns: ReadonlyArray<string>, logAboutUselessPatterns = true): Promise<string[]> {
+    const fileSet = new Set<string>();
+    for (const pattern of patterns) {
+      if (pattern.startsWith(IGNORE_PATTERN_CHARACTER)) {
+        const files = await this.expandPattern(pattern.substr(1), logAboutUselessPatterns);
+        files.forEach(fileName => fileSet.delete(fileName));
+      } else {
+        const files = await this.expandPattern(pattern, logAboutUselessPatterns);
+        files.forEach(fileName => fileSet.add(fileName));
+      }
+    }
+    return Array.from(fileSet);
+  }
+
+  private async expandPattern(globbingExpression: string, logAboutUselessPatterns: boolean): Promise<string[]> {
+    const fileNames = (await glob(globbingExpression)).map(relativeFile => path.resolve(relativeFile));
+    if (!fileNames.length && logAboutUselessPatterns) {
+      this.log.warn(
```

Comment:
Nice feature :-)

Reply:
Yeah, i was testing the new zero config out and didn't like a lot of "warnings" that weren't really warnings. However, I _do_ like the warning when a user configured it.